packages/@vue/cli-plugin-unit-jest/presets/default/jest-preset.js

```diff
@@ -1,11 +1,30 @@
-let vueJest = null
+// eslint-disable-next-line node/no-extraneous-require
+const semver = require('semver')
+
+let vueVersion = 2
 try {
-  vueJest = require.resolve('vue-jest')
-} catch (e) {
-  throw new Error('Cannot resolve "vue-jest" module. Please make sure you have installed "vue-jest" as a dev dependency.')
+  // eslint-disable-next-line node/no-extraneous-require
+  const Vue = require('vue')
+  vueVersion = semver.major(Vue.version)
+} catch (e) {}
```

Comment:
As we need to load a different transformer for Vue 2 and Vue 3, I hacked this code to retrieve the Vue version.
This might not be the best way, let me know what to do.

Reply:
You can require `vue/package.json` and check its `version` field